programs/survival/tputils.sc

```diff
@@ -0,0 +1,279 @@
+        return();
+    );
+
+    if(global_wdata:name==null,
+        print(player,format('r \nThe specefied location does not exist. Please use warp menu to know about current warps'));
+        return();
+    );
+    delete(global_wdata,name);
+    print(player,format(str('e \n%s was succesfully removed from warp list.',name)));
+
+);
+
+_wtp(name) -> (
+
+    player=player();
+    info = global_wdata:name;
+    if(info==null,
+        print(player,format('r \nThe specefied location does not exist. Please use warp menu to know about current warps'));
+        return();
+    );
+    dimension = info:'dimension';
+    pitch = info:'pitch';
+    yaw = info:'yaw';
+    location = info:'location';
+
+    run(str('execute in %s run tp %s %f %f %f %f %f',dimension,player,location:0,location:1,location:2,yaw,pitch));
+
+);
+
+_wmenu() -> (
+
+    player= player();
+    if(_screenCheck(),
+        warpCount = length(keys(global_wdata));
+        rowCount = ceil(warpCount/9);
+
+        menu= create_screen(player,str('generic_9x%s',rowCount),format('tb Available Warps'),_(screen,player,action,data) -> (
+            
+            if(action=='pickup' || action== 'swap' || action=='quick_move' || action== 'throw' || action== 'pickup_all',
+                itemNbt= inventory_get(screen,data:'slot'):2;
+                itemName= parse_nbt(itemNbt:'display':'Name'):0:'text';
+                _wtp(itemName);
+                close_screen(screen);
+            );
+        ));
+
+        if(screen_property(menu,'open'),
+
+            slot=0;
+            for(keys(global_wdata),
+
+                info = global_wdata:_;
+                iType=info:'item';
+                if(iType==null,iType='barrier');
+                des=info:'description';
+                loc=info:'location';
+                [x, y, z] = map(loc, floor(_));
+                dim= info:'dimension';
+                if(dim=='the_end',
+                    color='magenta',
+                //if else nether
+                dim=='the_nether',
+                    color='red',
+                //else
+                    color='green'
+                );
+                item=str('%s{display:{Name:\'[{"text":"%s","color":"%s","italic":false}]\',Lore:[\'[{"text":"Location: %s %s %s","color":"blue","italic":false}]\',\'[{"text":"Description:","color":"white","italic":false}]\',\'[{"text":"%s","italic":false}]\']}}',iType,str(_),color,x,y,z,des);
+                inventory_set(menu,slot,1,item);
+                slot+=1;
+            );
+        
+        );
+
+    );
+
+
+);
+
+
+//utility
+_screenCheck() -> (
+    sVersion= split('\\+',system_info('scarpet_version')):0; //fetches first part before the +, eg 1.4.57
+    vNumbers=split('\\.',sVersion);
+    [main, sub, x] = map(vNumbers,number(_));
+    if(main>1, 
+  true,
+  main == 1,
+  if(sub>4,
+    true,
+    sub==4,
+    x >= 56,
+    false
+  ),
+  false
+)
+);

```

Comment:
Suggestion for if this is false just don't register the subcommand, else a command that does nothing is weird.